### PR TITLE
[SYCL][PI] Add some fixes relevant to adapters second bump PR

### PR DESCRIPTION
**Pull request by omarahmed1111**
This PR add some fixes that is needed after adapters branch second bump PR, it needs only to me merged after the second bump [PR](https://github.com/oneapi-src/unified-runtime/pull/1014) for adapters branch is merged.

Depends on this [PR](https://github.com/oneapi-src/unified-runtime/pull/1014)

### CODE REVIEW

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -53,14 +53,14 @@ endif()
...
+  # commit be53fb3ba5bf1ac33051456d72c61b6a01d94a72
   # Author: Kenneth Benzie (Benie) <k.benzie@codeplay.com>
-  # Date:   Wed Nov 8 13:32:46 2023 +0000
-  #     Merge pull request #1022 from 0x12CC/l0_usm_error_checking_2
-  #     [UR][L0] Propagate OOM errors from `USMAllocationMakeResident`
-  set(UNIFIED_RUNTIME_TAG ec7982bac6cb3a6b9ed610cd6b7cb41fcbc780dc)
+  # Date:   Tue Oct 31 13:22:52 2023 +0000
+  # - Merge pull request #932 from Bensuo/cuda-cmd-buffers
+  # - [CUDA][EXP] CUDA adapter support for command buffers
+  set(UNIFIED_RUNTIME_TAG 7a95b06a762e2c4d1f4d2f76ba8cc1d2d0b09584)
```
Comment (fabiomestre) [inline sycl/plugins/unified_runtime/CMakeLists.txt:63]: I'm assuming this will be updated with the revision from https://github.com/oneapi-src/unified-runtime/pull/1014 right?

Comment (omarahmed1111) [reply sycl/plugins/unified_runtime/CMakeLists.txt]: Yes, this is just a temporary part until the other PR is merged.

File: sycl/plugins/unified_runtime/pi2ur.hpp [hunk 2]
```diff
@@ -2336,8 +2331,18 @@ inline pi_result piKernelGetInfo(pi_kernel Kernel, pi_kernel_info ParamName,
     break;
   }
   case PI_KERNEL_INFO_NUM_ARGS: {
-    UrParamName = UR_KERNEL_INFO_NUM_ARGS;
-    break;
+    size_t NumArgs = 0;
```
Comment (jandres742) [inline sycl/plugins/unified_runtime/pi2ur.hpp:2334]: why do we need this change? this seems like something it should go in a completely separate patch so in case there are regressions it is easy to identify.

Comment (aarongreig) [reply sycl/plugins/unified_runtime/pi2ur.hpp:2334]: without this one of the fixes we're pulling in causes a regression

Comment (omarahmed1111) [reply sycl/plugins/unified_runtime/pi2ur.hpp]: This changes was temporarily until this [PR](https://github.com/intel/llvm/pull/11794) is merged. This changes have been reverted and the PR was rebased on recent SYCL branch.

**Comment by omarahmed1111** [comment]
Due to a windows failure that will be fixed [here](https://github.com/oneapi-src/unified-runtime/pull/1084) and to make it easier, we will combine the second and third bump into one [here](https://github.com/oneapi-src/unified-runtime/pull/1072). So closing this for now.
